src/fides/api/ops/service/connectors/saas_connector.py

```diff
@@ -427,56 +432,76 @@ def run_consent_request(
         policy: Policy,
         privacy_request: PrivacyRequest,
         identity_data: Dict[str, Any],
-        consent_preferences: List[Consent],
     ) -> bool:
         """Execute a consent request. Return whether the consent request to the third party succeeded.
-        Return True if 200 OK
+        Should only propagate either the entire set of opt in or opt out requests.
+        Return True if 200 OK. Raises a SkippingConsentPropagation exception if no action is taken
+        against the service.
         """
         logger.info(
             "Starting consent request for node: '{}'",
             node.address.value,
         )
         self.set_privacy_request_state(privacy_request, node)
         query_config = self.query_config(node)
+        should_opt_in: Optional[bool] = should_opt_in_to_service(
+            self.configuration.system, privacy_request
+        )
```

Comment:
I'm having the saas connector collapse all the privacy preferences and/or consent preferences (old style) into a single optional boolean.  if True, we "opt in". If False, we "opt out". If "None" we do nothing.  see `should_opt_in_to_service` util which uses the same `filter_privacy_preferences_for_propagation` that the consent email connector uses in determining whether a preference is relevant for the current connector.